### ISSUE TITLE
Importing "rambda/immutable" works for types, but fails at build time

### ISSUE THREAD

**Issue by aisamu** [selfrefactor/rambda#585]
Although the types work (e.g. we see `readonly` types on the function signatures), trying to compile a typescript file fails with:

```
Cannot find module 'rambda/immutable'
```

[Node honors the "main" entry on `package.json`]( https://stackoverflow.com/questions/33621978/when-doing-require-for-npm-package-what-is-meaning-of-slash-in-package-name) when using "root" requires (e.g. `rambda`), but apparently ceases to do so when using "nested" ones (e.g. `rambda/immutable`), trying to refer instead to a file named `immutable.js`.

Inspecting the npm package shows that the file indeed doesn't exist.

-----

Creating a shim file named `immutable.js` that exported the contents of `./dist/rambda.js` fixed things:

#### `./immutable.js`

```js
module.exports = require('./dist/rambda.js')
```

---- 

Alternatively, creating a _folder_ named `immutable` and placing both the immutable declaration and the shim file there (while renaming them to `index`) also worked:

#### `tree .`

```
rambda
...
├── dist
│  ├── rambda.js
...
├── immutable
│  └── index.js
│  └── index.d.ts
│
├── index.d.ts
├── package.json
...
```

#### `./immutable/index.js`

```js
module.exports = require('../dist/rambda.js')
```

----
    
Alternatively (please bear with me!), I've also confirmed that a _folder_ named `immutable` with a `package.json` containing the alternative `main` and `typings` works.
This idea came up while reading Node's [require documentation](https://nodejs.org/docs/v0.4.2/api/modules.html#all_Together...).

#### `tree .`

```
rambda
...
├── dist
│  ├── rambda.js
...
├── immutable
│  └── package.json
│
├── index.d.ts
├── package.json
...
```

#### `./immutable/package.json`

```json
{
  "main": "../dist/rambda",
  "typings": "../immutable.d.ts"
}
```

> N.B.: I'm unaware of further implications of such an empty-looking package file (i.e. does it read missing entries from the parent `package.json`?)

----

I tried making these changes myself, but I'm afraid I failed to fully grasp the repo's build process.

Thank you very much for taking the time to write this awesome lib!

**Comment by selfrefactor** [comment]
Thank you for the report and for all the suggestions. I think I'll choose the first one with `./immutable.js`. I'll write again once new release with the fix is deployed.

**Comment by selfrefactor** [comment]
I applied the first suggestion to the new Rambdax version `7.3.0`. Can you confirm that it works, so I can apply the solution to Rambda?

**Comment by aisamu** [comment]
Confirmed that it works with `rambdax@7.3.0`! 🥳 
Thanks for the promptness!

**Comment by selfrefactor** [comment]
The fix is published to Rambda with version `6.8.3`. Thank you for bringing it up.
